File: backend/src/main/java/com/matchy/service/RegistrationService.java
package com.matchy.service;

import com.matchy.dto.RegistrationCreateDTO;
import com.matchy.dto.RegistrationDTO;
import com.matchy.entity.RegistrationStatus;

import java.util.List;

public interface RegistrationService {
    
    RegistrationDTO createRegistration(RegistrationCreateDTO createDTO);
    
    RegistrationDTO getRegistrationById(Long id);
    
    List<RegistrationDTO> getAllRegistrations();
    
    List<RegistrationDTO> getRegistrationsByEvenementId(Long evenementId);
    
    List<RegistrationDTO> getRegistrationsByUserId(Long userId);
    
    List<RegistrationDTO> getRegistrationsByStatus(RegistrationStatus status);
    
    RegistrationDTO approveRegistration(Long id);
    
    RegistrationDTO rejectRegistration(Long id);
    
    void deleteRegistration(Long id);
    
    Long getApprovedCountByEvenementId(Long evenementId);
}

File: src/app/core/services/registration.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Registration, RegistrationCreate, RegistrationStatus } from '../models/registration.model';

@Injectable({
  providedIn: 'root'
})
export class RegistrationService {
  private apiUrl = 'http://localhost:8081/api/registrations';

  constructor(private http: HttpClient) {}

  createRegistration(registration: RegistrationCreate): Observable<Registration> {
    return this.http.post<Registration>(this.apiUrl, registration);
  }

  getRegistrationById(id: number): Observable<Registration> {
    return this.http.get<Registration>(`${this.apiUrl}/${id}`);
  }

  getAllRegistrations(): Observable<Registration[]> {
    return this.http.get<Registration[]>(this.apiUrl);
  }

  getRegistrationsByEvenement(evenementId: number): Observable<Registration[]> {
    return this.http.get<Registration[]>(`${this.apiUrl}/evenement/${evenementId}`);
  }

  getRegistrationsByUser(userId: number): Observable<Registration[]> {
    return this.http.get<Registration[]>(`${this.apiUrl}/user/${userId}`);
  }

  getRegistrationsByStatus(status: RegistrationStatus): Observable<Registration[]> {
    return this.http.get<Registration[]>(`${this.apiUrl}/status/${status}`);
  }

  approveRegistration(id: number): Observable<Registration> {
    return this.http.put<Registration>(`${this.apiUrl}/${id}/approve`, {});
  }

  rejectRegistration(id: number): Observable<Registration> {
    return this.http.put<Registration>(`${this.apiUrl}/${id}/reject`, {});
  }

  deleteRegistration(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }

  getApprovedCount(evenementId: number): Observable<number> {
    return this.http.get<number>(`${this.apiUrl}/evenement/${evenementId}/approved-count`);
  }
}
